lib/CMakeLists.txt

```diff
@@ -75,20 +75,23 @@ set(PROJECT_SOURCES
 ## Without the intermediate target, unexported symbols can't be tested.
 add_library(libprojectsrc OBJECT ${PROJECT_SOURCES})
 set_target_properties(libprojectsrc PROPERTIES POSITION_INDEPENDENT_CODE true)
-
-add_library(lib${PROJECT_NAME} $<TARGET_OBJECTS:libprojectsrc>)
-set_target_properties(lib${PROJECT_NAME} PROPERTIES VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
-target_link_libraries(lib${PROJECT_NAME}
+if(CPP_HOCON_SHARED)
```

Comment:
This is actually unnecessary, you can use https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html